lib/delayed/backend/active_record.rb

```diff
@@ -109,11 +109,18 @@ def self.reserve_with_scope_using_optimized_sql(ready_scope, worker, now)
         end
 
         def self.reserve_with_scope_using_default_sql(ready_scope, worker, now)
-          # This is our old fashion, tried and true, but slower lookup
-          ready_scope.limit(worker.read_ahead).detect do |job|
-            count = ready_scope.where(id: job.id).update_all(locked_at: now, locked_by: worker.name)
-            count == 1 && job.reload
+          # This is our old fashion, tried and true, but possibly slower lookup
+          # Instead of reading the entire job record for our detect loop, just pluck the ID,
+          # and only read the job record after we've successfully locked the job.
+          # This can be particularly helpful when operating a large job cluster that uses
+          # a large read_ahead value to increase the odds of successfully locking a job with
+          # one method call to reserve_with_scope.
+          locked_job_id = ready_scope.limit(worker.read_ahead).pluck(:id).detect do |job_id|
+            count = ready_scope.where(id: job_id).update_all(locked_at: now, locked_by: worker.name)
+            count == 1
           end
+
+          where(id: locked_job_id).first if locked_job_id
```

Comment:
This will add an unnecessary ORDER BY clause to the query. Could you use `find_by` instead?

Reply:
Good catch! I also just learned you can use `.take` instead of `.first` to eliminate the implicit ordering, but I think your suggestion is more readable.